### DISCUSSION TITLE
Not able to delete a list of files

### DISCUSSION THREAD

**Discussion by aborruso** [nushell/nushell#10933] [Q&A] [answered]
Hi,
if I run

```
ls -l `C:\Users\aborr\Desktop\trash\**\*` | where created <= (date now) - 180day and type == file
```

I have a list of files.

If I add the rm of each

```
ls -l `C:\Users\aborr\Desktop\trash\**\*` | where created <= (date now) - 180day and type == file |  each { rm -fv $in.name }
```

and then run again 

```
ls -l `C:\Users\aborr\Desktop\trash\**\*` | where created <= (date now) - 180day and type == file
```

I have the same list of files, as if nothing had been cancelled.

What's wrong in the way I delete the files?

I'm using nu 0.80.0.

Thank you

**Comment by amtoine** [comment]
if the `trash/` directory is where the files go when they are deleted, then you might want to try with `rm --permanent` :+1:

**Reply by aborruso** [reply]
> if the `trash/` directory is where the files go when they are deleted, then you might want to try with `rm --permanent` 👍

No it's not the directory where the files go when they are deleted, it's a normal directory. But probably it's because I have strange file names.

In example if I run

```
rm "[Italian (auto-generated)] lorem ipsum [DownSub.com].srt"
```

I have

```
Error:
  × File(s) not found
   ╭─[entry #12:1:1]
 1 │ rm "[Italian (auto-generated)] lorem ipsum [DownSub.com].srt"
   ·    ─────────────────────────────┬────────────────────────────
   ·                                 ╰── File(s) not found
   ╰────
```

**Reply by amtoine** [reply]
ahh yeah that might be it

`rm` is probably trying to expand the filename as a *glob* pattern :thinking:

**Reply by aborruso** [reply]
And how to process the output of ls command, to rm these file with strange names?

**Reply by amtoine** [reply]
i'm not sure, i never have such filenames...
feel free to drop in our Discord, you might be more lucky there :relieved:

**Reply by fdncred** [reply]
You should upgrade to 0.86.0 and see if it's still broken. We've had some PRs in this area for file names that have glob syntax in the file name or path.

**Reply by aborruso** [reply]
> You should upgrade to 0.86.0 and see if it's still broken. We've had some PRs in this area for file names that have glob syntax in the file name or path.


@fdncred thank you. Using 0.86.0 I have the same result

```
Error:   × File(s) not found
   ╭─[entry #3:1:1]
 1 │ rm "[Italian (auto-generated)] lorem ipsum [DownSub.com].srt"
   ·    ─────────────────────────────┬────────────────────────────
   ·                                 ╰── File(s) not found
   ╰────
```

**Reply by fdncred** [reply, marked as answer]
Ah, I think I see why https://github.com/nushell/nushell/pull/10694 was landed shortly after the release. So, you can either try compiling the main branch, use a nightly https://github.com/nushell/nightly/releases or wait for 0.87.0.

**Reply by aborruso** [reply]
I will wait for 0.87. Thank you @fdncred
